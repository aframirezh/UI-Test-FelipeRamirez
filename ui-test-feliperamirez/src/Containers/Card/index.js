import React, { Component } from 'react';
import './styles.scss';
import Postulates from '../../Components/Postulates';
 
class Card extends Component  {
 
   constructor(props) {
       super(props);
       this.state = {
           profilesInfo: []
       };
     }

   render (){
       const info = this.props;
       const containerClass = info.cardFlag ? "card-container--header":"card-container";
       const containerBackgroundClass = "card-container_background card-container_background--"+ info.position;
       return (
               <div className={containerClass}>
                   <div className={ info.cardFlag ? " card-container_background--header" : containerBackgroundClass}>
                       <div className="grid box" >
                          <Postulates key={info.id} {...info}/>
                       </div>
                       <div className="grid box--good">
                           <span className="oi oi-thumb-up"></span>
                       </div>
                       <div className="grid box--bad">
                           <span className="oi oi-thumb-down box-bad-down"></span>
                       </div>
                   </div>
               </div>
       );
   }
}
 
export default Card;