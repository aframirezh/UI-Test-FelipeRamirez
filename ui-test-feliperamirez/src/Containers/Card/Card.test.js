import React from 'react';
import Card from '.';
import renderer from 'react-test-renderer';
describe('Card rendering', () => {
    it('Card correctly rendering', () => {
        const props = { 
            cardFlag: false, 
            id: '1',
            name: 'Test Name',
            description: 'Description test', 
            position:1,
        };
        const component = renderer.create(
            (<Card {...props}/>),
        );
        const card = component.toJSON();

        expect(card).toMatchSnapshot();
    })
})