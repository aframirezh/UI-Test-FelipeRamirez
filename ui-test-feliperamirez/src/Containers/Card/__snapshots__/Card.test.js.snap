// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card rendering Card correctly rendering 1`] = `
<div
  className="card-container"
>
  <div
    className="card-container_background card-container_background--1"
  >
    <div
      className="grid box"
    >
      <div>
        <section>
          <p>
            <strong
              className="box-tittle-votes-size regular"
            >
              Test Name
            </strong>
          </p>
          <p
            className="thin"
          />
          <p
            className="regular"
          >
            Description test
          </p>
          <div>
            <div
              className="middle"
            >
              <label
                className="order-left"
              >
                <input
                  name="radio"
                  type="radio"
                  value="good"
                />
                <div
                  className="vote-good box-vote"
                >
                  <span
                    className="oi oi-thumb-up"
                  />
                </div>
              </label>
              <label
                className="order-left"
              >
                <input
                  name="radio"
                  type="radio"
                  value="bad"
                />
                <div
                  className="good-bad box-vote"
                >
                  <span
                    className="oi oi-thumb-down box-bad-down"
                  />
                </div>
              </label>
              <label
                className="order-left"
              >
                <button
                  className="button"
                  onClick={[Function]}
                >
                  Vote now
                </button>
              </label>
            </div>
          </div>
        </section>
      </div>
    </div>
    <div
      className="grid box--good"
    >
      <span
        className="oi oi-thumb-up"
      />
    </div>
    <div
      className="grid box--bad"
    >
      <span
        className="oi oi-thumb-down box-bad-down"
      />
    </div>
  </div>
</div>
`;
